### PR TITLE
Implement prisoner payment details form design

### PR DESCRIPTION
**Pull request by ushkarev**

### CODE REVIEW

File: mtp_send_money/apps/send_money/fields.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        super().__init__(fields, *args, **kwargs)
+
+    def compress(self, data_list):
+        if data_list:
+            try:
+                if any(item in self.empty_values for item in data_list):
+                    raise ValueError
+                return datetime.date(data_list[2], data_list[1], data_list[0])
+            except ValueError:
+                raise ValidationError(_('Enter a valid date.'))
```
Comment (maxf) [inline mtp_send_money/apps/send_money/fields.py]: We need all the error messages to be in the templates, or at least away from the code. Possible?

File: mtp_send_money/assets-src/javascripts/modules/charges.js [hunk 2]
```diff
@@ -11,27 +11,32 @@ exports.Charges = {
...
-    return num.toFixed(2).toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+    return '£' + num.toFixed(2).toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
   },
 
-  _updateTotal: function (event) {
-    var amount = event.target.value;
-    var serviceCharge;
-    if (/^\d*\.?\d{2}?$/.test(amount)) {
+  _updateTotal: function (amount) {
+    if (/^ *\d+(\.\d{2})? *$/.test(amount)) {
```
Comment (ushkarev) [inline mtp_send_money/assets-src/javascripts/modules/charges.js]: perhaps do `amount.trim()` to keep the regular expression simpler and also remove other forms of whitespace?

File: mtp_send_money/apps/send_money/templatetags/send_money.py [hunk 3]
```diff
@@ -1,22 +1,48 @@
...
+def add_service_charge(amount):
+    return get_total_charge(amount)
 
 
 @register.filter
-def currency_format(amount):
-    if not isinstance(amount, decimal.Decimal):
-        amount = unserialise_amount(amount)
-    return '£' + serialise_amount(amount)
+def get_widget_attrs(bound_field, field_index=None):
```
Comment (ibrechin) [inline mtp_send_money/apps/send_money/templatetags/send_money.py]: Is this used anywhere?

Comment (ushkarev) [inline mtp_send_money/apps/send_money/templatetags/send_money.py]: it was when the type and browser-based validation attributes were added to form fields… it's a convenience function. shall i remove it as a fix-up to the _Revert: Use more specific input types to enable native browser validation_ commit?

Comment (ibrechin) [inline mtp_send_money/apps/send_money/templatetags/send_money.py]: sure

**Pull request merged**